src/utils/streams/promise_from_streams.js

```diff
@@ -0,0 +1,69 @@
+import { race } from 'bluebird';
+
+/**
+ *  Take an array of streams, pipe the output
+ *  from each one into the next, listening for
+ *  errors from any of the streams, and then resolve
+ *  the promise once the final stream has finished
+ *  writing/reading.
+ *
+ *  If the last stream is readable, it's final value
+ *  will be provided as the promise value.
+ *
+ *  Errors emmitted from any stream will cause
+ *  the promise to be rejected with that error.
+ *
+ *  @param  {Array<Stream>} streams
+ *  @return {Promise<undefined>}
```

Comment:
The `undefined` doesn't match the docs above: "will be provided as the promise value"